Tour-Collective-Devs/tc_backend
# Refactor auth

## Link to Ticket

Autolink format: hashtag and number, ex: #

Closes #12

## Description of Proposed Changes
Refactored auth views to use custom written methods rather than the package we were trying to use before. Also moved the user model and other user classes into the api app and removed the old user app.

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout refactor-auth
```
1. pull down code
1. completely remove all migrations and databases and remake them
1. run makemigrations for api and migrate into a new database
1. run server and open up postman, unfortunately these new views do not appear in the browsable api.
1. post a user object to the endpoint at /api/auth/register/
1. you should see a return of a user token
1. post login information to the endpoint at /api/auth/login/
1. you should once again see a return of a user token

example register information json
```
{
	"username": "testuser",
	"password": "lolhacked",
	"email": "test@user.com",
	"first_name": "test",
	"last_name": "test",
	"is_employer": false,
	"is_crew_member": true
}
```

example login json
```
{
	"username": "testuser",
	"email": "test@user.com",
	"password": "lolhacked"
}
```

## Impacted Areas in Application
api, users, auth, login, register

List general components of the application that this PR will affect:

## Mentions @username

Tag users that need to review this code

## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Deploy Notes

Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

## Definition of Done

1. The project must be fully documented. This includes the following:
1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
1. Every class must be documented with purpose, author, and methods.
1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
1. Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
